section2/methods.rb

```diff
@@ -23,35 +19,56 @@ def add(num2, num1)
 end
 
 def modulus(num1, num2)
-  if num2.to_f.eql?(0.0)
-    return "invalid, as #{num1} could not be modulus by #{num2}" # The first number modulus the second number is invalid, as 1 could not perform modulus by 0
-  end
-
-  return num1.to_f % num2.to_f
+  num1.to_f % num2.to_f
 end
 
 puts "Simple Calculator"
 20.times { print "-"}
 puts 
 
+# Declare Variable
+$num_1 
+$num_2
+
 #1
-puts "Please enter the first number" 
-num_1 = gets.chomp
-puts "Please enter the second number"
-num_2 = gets.chomp
+loop do
+  puts "Please enter the first number" 
+  $num_1 = gets.chomp
+
+  puts "Please enter the second number"
+  $num_2 = gets.chomp
+
+  begin
+    # To check if the input is Number
+    Float($num_1)
+    Float($num_2)
+
+    # Check division or modulus able to perform by checking second number
+    if ($num_2.to_f.eql?(0.0))
```

Comment:
If you do the above, then this could just be:

```suggestion
    if ($num_2.eql?(0))
```